optuna/visualization/_parallel_coordinate.py

```diff
@@ -173,47 +230,81 @@ def _target(t: FrozenTrial) -> float:
             else:
                 values = [vocab[v] for v in values]
                 ticktext = list(sorted(vocab.keys(), key=lambda x: vocab[x]))
-
-            dim = {
-                "label": p_name if len(p_name) < 20 else "{}...".format(p_name[:17]),
-                "values": tuple(values),
-                "range": (min(values), max(values)),
-                "tickvals": list(range(len(vocab))),
-                "ticktext": ticktext,
-            }
+            dim = _DimensionInfo(
+                label=_truncate_label(p_name),
+                values=tuple(values),
+                range=(min(values), max(values)),
+                is_cat=True,
+                tickvals=list(range(len(vocab))),
+                ticktext=ticktext,
+            )
         else:
-            dim = {
-                "label": p_name if len(p_name) < 20 else "{}...".format(p_name[:17]),
-                "values": tuple(values),
-                "range": (min(values), max(values)),
-            }
+            dim = _DimensionInfo(
+                label=_truncate_label(p_name),
+                values=tuple(values),
+                range=(min(values), max(values)),
+            )
 
         dims.append(dim)
 
     if numeric_cat_params_indices:
+        dims.insert(0, dim_objective)
```

Comment:
[nits] you do not need to insert it here, if you use start=0.

Reply:
The dim_objective cannot calculated with the for block, so we need to add it here., i think.